Sources/DateExtensions.swift

```diff
@@ -213,41 +242,51 @@ extension Date {
     public var year: Int {
         return Calendar.current.component(Calendar.Component.year, from: self)
     }
-
+    
     /// EZSE : Get the month from the date
     public var month: Int {
         return Calendar.current.component(Calendar.Component.month, from: self)
     }
-
+    
     /// EZSE : Get the weekday from the date
     public var weekday: String {
-        let dateFormatter = DateFormatter()
-        dateFormatter.dateFormat = "EEEE"
-        return dateFormatter.string(from: self)
+        let format = "EEEE"
+        if let dateFormatter = dateFormatterDict[format]{
```

Comment:
This guarded check is used in a whole bunch of places. Can you please make a private method to put the logic for getting the date formattter in one place?

Reply:
I have updated the pull request kindly review @Khalian 
I have created a method for getting formatters from dict.